streams/test-utils/src/main/java/org/apache/kafka/streams/test/ConsumerRecordFactory.java

```diff
@@ -215,87 +254,183 @@ public void advanceTimeMs(final long advanceMs) {
+     * Does not auto advance internally tracked time.
+     *
+     * @param value the record value
+     * @param headers the record headers
+     * @param timestampMs the record timestamp
+     * @return the generated {@link ConsumerRecord}
+     */
+    public ConsumerRecord<byte[], byte[]> create(final V value,
+                                                 final long timestampMs,
+                                                 final Headers headers) {
         if (topicName == null) {
             throw new IllegalStateException("ConsumerRecordFactory was created without defaultTopicName. " +
                 "Use #create(String topicName, V value, long timestampMs) instead.");
         }
-        return create(topicName, value, timestampMs);
+        return create(topicName, value, headers, timestampMs);
+    }
+
+    /**
+     * Create a {@link ConsumerRecord} with {@code null}-key and the given topic name, value and headers.
+     * The timestamp will be generated based on the constructor provided start time and time will auto advance.
+     *
+     * @param topicName the topic name
+     * @param value the record value
+     * @param headers the record headers
+     * @return the generated {@link ConsumerRecord}
+     */
+    public ConsumerRecord<byte[], byte[]> create(final String topicName,
+                                                 final V value,
+                                                 final Headers headers) {
+        return create(topicName, null, value, headers);
     }
 
     /**
      * Create a {@link ConsumerRecord} with {@code null}-key and the given topic name and value.
-     * The timestamp will be generated from the constructor provided and time will auto advance.
+     * The timestamp will be generated based on the constructor provided start time and time will auto advance.
      *
      * @param topicName the topic name
      * @param value the record value
      * @return the generated {@link ConsumerRecord}
      */
     public ConsumerRecord<byte[], byte[]> create(final String topicName,
                                                  final V value) {
-        return create(topicName, null, value);
+        return create(topicName, null, value, new RecordHeaders());
     }
 
     /**
      * Create a {@link ConsumerRecord} with default topic name and {@code null}-key was well as given value.
-     * The timestamp will be generated from the constructor provided and time will auto advance.
+     * The timestamp will be generated based on the constructor provided start time and time will auto advance.
      *
      * @param value the record value
      * @return the generated {@link ConsumerRecord}
      */
     public ConsumerRecord<byte[], byte[]> create(final V value) {
+        return create(value, new RecordHeaders());
+    }
+
+    /**
+     * Create a {@link ConsumerRecord} with default topic name and {@code null}-key was well as given value.
```

Comment:
nit: header is missing -> `given value[ and headers].`

Reply:
ack..